packages/typespec-client-generator-core/src/public-utils.ts

```diff
@@ -120,7 +127,11 @@ export function getClientNamespaceString(context: TCGCContext): string | undefin
  * @param type
  * @returns
  */
-export function getEffectivePayloadType(context: TCGCContext, type: Model): Model {
+export function getEffectivePayloadType(
+  context: TCGCContext,
+  type: Model,
+  visibility?: Visibility,
+): Model {
```

Comment:
Update the docstring?

I am not sure whether `Visibility.Read` mean here. In the test, `Visibility.Create` would be not affect the effective type. Then what kind of property would affect? `@visibility(Visibility.Read) prop`?

Reply:
doc updated. the visibility here means only take the properties with the visibility provided into account when finding the potential named model.

Reply:
So, the test case works, because the response from http lib is
```
{
  read: string
}
```
and via this `getEffectivePayloadType(..., Visibility.READ)`, it would map to `Test` model?

Reply:
correct.